invenio_rdm_records/resources/serializers/datacite/schema.py

```diff
@@ -373,12 +374,45 @@ def get_locations(self, obj):
             if geometry:
                 geo_type = geometry["type"]
                 # PIDS-FIXME: Scalable enough?
-                # PIDS-FIXME: Implement Box and Polygon serialization
                 if geo_type == "Point":
                     serialized_location["geoLocationPoint"] = {
-                        "pointLatitude": geometry["coordinates"][0],
-                        "pointLongitude": geometry["coordinates"][1],
+                        "pointLongitude": str(geometry["coordinates"][0]),
+                        "pointLatitude": str(geometry["coordinates"][1]),
                     }
+                if geo_type == "Polygon":
```

Comment:
minor: to highlight we are in a different case (+ remove conditional check)

```suggestion
                elif geo_type == "Polygon":
```